### PR TITLE
    [msan] Convert vector shadow to scalar before zext

### PR DESCRIPTION
**Pull request by thurstond**
zext does not allow converting vector shadow to scalar, so we must manually convert it prior to calling zext in materializeOneCheck, for which the 'ConvertedShadow' parameter isn't actually guaranteed to be scalar (1). Note that it is safe/no-op to call convertShadowToScalar on a shadow that is already scalar.

In contrast, the storeOrigin function already converts the (potentially vector) shadow to scalar; we add a comment to note why it is load bearing.

(1) In materializeInstructionChecks():
"// Disable combining in some cases. TrackOrigins checks each shadow to pick
 // correct origin.
 bool Combine = !MS.TrackOrigins;
 ...
       if (!Combine) {
        materializeOneCheck(IRB, ConvertedShadow, ShadowData.Origin);
        continue;
      }"

### CODE REVIEW

File: llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp [hunk 1]
```diff
@@ -1398,6 +1399,8 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     if (instrumentWithCalls(ConvertedShadow) &&
         SizeIndex < kNumberOfAccessSizes && !MS.CompileKernel) {
       FunctionCallee Fn = MS.MaybeWarningFn[SizeIndex];
+      // ZExt cannot convert between vector and scalar
+      ConvertedShadow = convertShadowToScalar(ConvertedShadow, IRB);
```
Comment (vitalybuka) [inline llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp:1403]: we need IR test

File: llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp [hunk 2]
```diff
@@ -1283,6 +1283,7 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
     const DataLayout &DL = F.getParent()->getDataLayout();
     const Align OriginAlignment = std::max(kMinOriginAlignment, Alignment);
     TypeSize StoreSize = DL.getTypeStoreSize(Shadow->getType());
+    // ZExt cannot convert between vector and scalar
     Value *ConvertedShadow = convertShadowToScalar(Shadow, IRB);
```
Comment (vitalybuka) [inline llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp:1287]: I don't know why we check isZero on coverted shado
I'd rather expect after that.

But this is not for this patch.

**Review by vitalybuka — CHANGES_REQUESTED** [review]
Tests is needed

File: llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp (hunk 1 above)
Comment (thurstond) [reply llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp:1403]: Baseline IR tests at https://github.com/llvm/llvm-project/pull/97611

I'll update this PR to show the corrected IR output

**Review by vitalybuka — APPROVED** [review]

**Pull request merged**